plugins/modules/host.py

```
@@ -0,0 +1,284 @@
+            if existing:
+                self.host_output = existing
+            else:
+                self.check_required_host_id_param()
+                body = {"items": [{"hostname": self.host_name,"ipAddress": self.host_ip }]}
+                create_host = host_api_instance.create_hosts(body=body).to_dict()
+                self.host_output = host_api_instance.read_host(host_id=create_host['items'][0]['host_id']).to_dict()
+                self.changed = True
+
+        if self.state in ['absent']:
+            if existing:
+                self.host_output = host_api_instance.delete_host(host_id=existing['host_id']).to_dict()
+                self.changed = True
+
+        if self.state in ['attach','detach']:
+            self.check_required_cluster_name_param()
+            try:
+                cluster_api_instance.read_cluster(cluster_name=self.cluster_name).to_dict()
+            except ApiException as ex:
+                if ex.status == 404:
+                    self.module.fail_json(msg=f"Cluster does not exist:  {self.cluster_name}")
+
+        if self.state in ['attach']:
+            if existing:
+                try:
+                    body = {"items": [{"hostId": existing['host_id'],"hostname": self.host_name }]}
+                    cluster_api_instance.add_hosts(cluster_name=self.cluster_name,body=body).to_dict()
+                    self.changed = True
+                except ApiException as ex:
+                    if ex.status == 400:
+                        pass
+                self.host_output = host_api_instance.read_host(host_id=host_id).to_dict()
+            else:
+                self.check_required_host_id_param()
+                body = {"items": [{"hostname": self.host_name,"ipAddress": self.host_ip }]}
+                create_host = host_api_instance.create_hosts(body=body).to_dict()
+                body = {"items": [{"hostId": create_host['items'][0]['host_id'],"hostname": self.host_name }]}
```

Use `ApiHostRefList` and `ApiHostRef` for fail fast